### PR TITLE
Introduced `TriggeredItem` to better abstract `getTriggers`

### PR DESCRIPTION
**Pull request by jglick**
Allows some old tech debt to be cleaned up in https://github.com/jenkinsci/cloudbees-folder-plugin/pull/264.

### Proposed changelog entries

- Introduced a new interface `TriggeredItem`.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7131"></a>

### CODE REVIEW

File: core/src/main/java/hudson/triggers/Trigger.java
Comment (jglick) [inline core/src/main/java/hudson/triggers/Trigger.java:285]: Not really related, but I noticed that https://github.com/jenkinsci/jenkins/blob/1df554abd76811fd58e6383a614e224700aeecaf/core/src/main/java/hudson/triggers/Trigger.java#L262-L266 did not apply to Pipeline projects, so turning on this flag would have skipped them in SCM polling.

Comment (Vlatombe) [reply core/src/main/java/hudson/triggers/Trigger.java:285]: Shouldn't the block you referenced be updated to use `TriggeredItem` as well and take a `Job` instead of an `AbstractProject` ?

Comment (jglick) [reply core/src/main/java/hudson/triggers/Trigger.java:285]: No, it really needs `AbstractProject`: https://issues.jenkins.io/browse/JENKINS-29913

**Review by Vlatombe — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
